### PR TITLE
Notify competitors that they should claim their WCA ID after a competition

### LINKED ISSUE #991: Notify competitors that they should claim their WCA ID
I just realized we don't have an automatic mail to remind competitors to claim their id: it could be useful so that they have it ready for their next competition (it saves time when validating the results spreadsheet).

Obviously we cannot know for sure which id belongs to who, however what we could do is, after the competition's results have just been posted, check all the registered competitors without a linked WCA id, and mail them that if they indeed took part to the competition they should claim their id.

How does it sound?
I actually just did that manually for the last competition I delegated, which is why it could be cool if it would be automatic :p

### PR DESCRIPTION
**Pull request by jonatanklosko**
Fixes #991.

Let me know if that's the way you thought of =)

### CODE REVIEW

**Comment by jfly** [comment]
Thanks for getting this discussion started, Jonatan! Could you write up exactly what the behavior of this is? As I understand it, this would notify everyone who registered, regardless of if they are on the waiting list, or if they actually showed up to compete?

**Comment by jonatanklosko** [comment]
That's a good point @jfly! I've just updated the code to notify accepted registrants only. We still cannot be sure if the person has shown up, but it seems that there's no much we can do about that. Maybe we could add something to the email clarifying the situation? (Not sure how to formulate this tho)

**Comment by ronaldmansveld** [comment]
I guess we could determine if people showed up based on the published
results? So only send out the email to people who have published results,
but an unclaimed ID?

**Comment by jonatanklosko** [comment]
So far I thought that wouldn't be 100% reliable system. We could determine them by name for example but names can slightly change in the results posting process, that's basically what #21 is about, @jfly am I right?

**Comment by jfly** [comment]
> So far I thought that wouldn't be 100% reliable system. We could determine them by name for example but names can slightly change in the results posting process, that's basically what #21 is about, @jfly am I right?

That's correct, #21 is about dealing with the fact that people's names change. Since that code is still PHP, I don't think it's worth trying to tackle it right now.

I think it's fine to email everyone, but we need to update the text of the email to make it very clear what's going on here. As @viroulep said in #991: "and mail them that if they indeed took part to the competition they should claim their id.". Right now, the email says "you competed in ...", we should update it to say something like "you registered for ... and it looks like it was your first time competing, if so, you should claim your wca id"

File: WcaOnRails/app/mailers/competitions_mailer.rb [hunk 1]
```diff
@@ -22,6 +22,14 @@ def notify_users_of_results_presence(user, competition)
     )
   end
 
+  def notify_users_of_id_claim_possibility(user, competition)
+    @competition = competition
+    mail(
+      to: user.email,
```
Comment (jfly) [inline WcaOnRails/app/mailers/competitions_mailer.rb]: Quick thought: people *love* replying to automated emails. Have you thought about a reasonable `reply_to` header for this email? Maybe the delegates of the competition?

Comment (jonatanklosko) [inline WcaOnRails/app/mailers/competitions_mailer.rb]: Yeah, cannot think of better `reply_to` in this case.

**Comment by KitClement** [comment]
Just saw the email about this feature in the delegate list -- is there any reason we don't automatically assign IDs to accounts when the use that account to register for their first competition?

**Comment by jfly** [comment]
> is there any reason we don't automatically assign IDs to accounts when the use that account to register for their first competition? -@KitClement 

Yeah, please read through #21

**Comment by jfly** [comment]
LGTM!

**Pull request merged**